src/dash-table/DataTable.js

```diff
@@ -157,6 +145,49 @@ export const propTypes = {
             PropTypes.number
         ]),
 
+        /**
+         * The formatting applied to the column's data.
+         *
+         * This prop is derived from the [d3-format](https://github.com/d3/d3-format) library specification. Apart from
+         * being structured slightly differently (under a single prop), the usage
+         * is the same.
+         *
+         * 'locale': represents localization specific formatting information
+         *   When left unspecified, will use the default value provided by d3-format.
+         *
+         *   'symbol': (default: ['', $']) a list of two strings representing the
+         *   prefix and suffix symbols
```

Comment:
default `['$', '']`

OK, I'm comfortable generalizing currency to symbol here - but let's describe what's happening, both to elucidate the common use case and to not confuse d3 aficionados. Perhaps:

a list of two strings representing the prefix and suffix symbols. Typically used for currency, and implemented using d3's currency format, but you can use this for other symbols such as measurement units.